matsim/src/main/java/org/matsim/analysis/pt/stop2stop/PtStop2StopAnalysis.java

```diff
@@ -225,132 +234,48 @@ private PtVehicleData(Id<TransitLine> transitLineId, Id<TransitRoute> transitRou
-            return linkIdsSincePreviousStop;
-        }
-    }
+	/**
+	 * output data structure
+	 *
+	 * @param transitLineId for aggregation -> set null or leave out and use Map lineId -> Stop2StopEntry?
+	 */
+	record Stop2StopEntry(Id<TransitLine> transitLineId, Id<TransitRoute> transitRouteId, Id<Departure> departureId, Id<TransitStopFacility> stopId,
+							  int stopSequence, Id<TransitStopFacility> stopPreviousId, double arrivalTimeScheduled, double arrivalDelay,
+							  double departureTimeScheduled, double departureDelay, double passengersAtArrival, double totalVehicleCapacity,
+							  double passengersAlighting, double passengersBoarding, List<Id<Link>> linkIdsSincePreviousStop) {}
 
     static final String[] HEADER = {"transitLine", "transitRoute", "departure", "stop", "stopSequence",
             "stopPrevious", "arrivalTimeScheduled", "arrivalDelay", "departureTimeScheduled", "departureDelay",
             "passengersAtArrival", "totalVehicleCapacity", "passengersAlighting", "passengersBoarding",
             "linkIdsSincePreviousStop"};
 
-    public static Comparator<Stop2StopEntry> stop2StopEntryByTransitLineComparator =
-            Comparator.nullsLast(Comparator.comparing(Stop2StopEntry::getTransitLineId));
-    public static Comparator<Stop2StopEntry> stop2StopEntryByTransitRouteComparator =
-            Comparator.nullsLast(Comparator.comparing(Stop2StopEntry::getTransitRouteId));
-    public static Comparator<Stop2StopEntry> stop2StopEntryByDepartureComparator =
-            Comparator.nullsLast(Comparator.comparing(Stop2StopEntry::getDepartureId));
-    public static Comparator<Stop2StopEntry> stop2StopEntryByStopSequenceComparator =
-            Comparator.nullsLast(Comparator.comparing(Stop2StopEntry::getStopSequence));
+    static Comparator<Stop2StopEntry> stop2StopEntryByTransitLineComparator =
+            Comparator.nullsLast(Comparator.comparing(Stop2StopEntry::transitLineId));
+    static Comparator<Stop2StopEntry> stop2StopEntryByTransitRouteComparator =
+            Comparator.nullsLast(Comparator.comparing(Stop2StopEntry::transitRouteId));
+    static Comparator<Stop2StopEntry> stop2StopEntryByDepartureComparator =
+            Comparator.nullsLast(Comparator.comparing(Stop2StopEntry::departureId));
+    static Comparator<Stop2StopEntry> stop2StopEntryByStopSequenceComparator =
+            Comparator.nullsLast(Comparator.comparing(Stop2StopEntry::stopSequence));
+
+	public void writeStop2StopEntriesByDepartureCsv(String fileName, String columnSeparator, String listSeparatorInsideColumn) {
+		writeStop2StopEntriesByDepartureCsv(IOUtils.getFileUrl(fileName), columnSeparator, listSeparatorInsideColumn);
+	}
+
+	public void writeStop2StopEntriesByDepartureCsv(Path path, String columnSeparator, String listSeparatorInsideColumn) {
+		writeStop2StopEntriesByDepartureCsv(IOUtils.getFileUrl(path.toString()), columnSeparator, listSeparatorInsideColumn);
```

Comment:
OutputOptions in contrib application provides only output paths as `Path` and IOUtils accepts only `String` and `URL`. This results in this ugly conversion here.